src/hotspot/share/cds/regeneratedClasses.hpp

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+#ifndef SHARE_CDS_REGENERATEDCLASSES_HPP
+#define SHARE_CDS_REGENERATEDCLASSES_HPP
+
+#include "memory/allStatic.hpp"
+#include "oops/oopHandle.hpp"
+#include "runtime/handles.hpp"
+#include "utilities/growableArray.hpp"
+#include "utilities/resourceHash.hpp"
+
+class RegeneratedClasses : public AllStatic {
+ private:
+  using RegeneratedObjTable = ResourceHashtable<address, address, 15889, AnyObj::C_HEAP, mtClassShared>;
+  static RegeneratedObjTable* _renegerated_objs; // InstanceKlass* and Method*
+  static GrowableArrayCHeap<OopHandle, mtClassShared>* _regenerated_mirrors;
```

Comment:
Since these are private, and not used by the argument, do they need to be in the header file?

Reply:
I moved them to the cpp file.